examples/generate_bom/generate_bom.py

```diff
@@ -137,87 +66,48 @@ def extract_attributes_from_schdoc(schdoc_file_content, attributes_to_extract):
-    prjpcb_branch = prjpcb_repo.get_branch(source_branch)
-    prjpcb_file = get_file_content(prjpcb_repo, source_file, prjpcb_branch)
-    schdoc_files_in_proj = {x for x in get_schdoc_list_from_prjpcb(prjpcb_file)}
-
-    print(f"{schdoc_files_in_proj=}")
-
-    # Get the SchDoc files from the SchDoc repo.
-    schdoc_repo = allspice.get_repository(schdoc_repo_owner, schdoc_repo_name)
-    schdoc_branch = schdoc_repo.get_branch(schdoc_branch)
-    schdoc_files_in_repo = schdoc_repo.get_git_content(ref=schdoc_branch)
-
-    # Extract attributes from the SchDoc files. See thhe
-    # `extract_attributes_from_schdoc` function for more information.
-    bom_rows = []
-
-    # Note: we're going through the files in the repo and checkin if they're
-    # in the PrjPcb file. This may miss some files in the proj if they're
-    # not in the repo. If you're seeing inconsistent or missing results,
-    # check that all the SchDoc files are in the repo, or uncomment this line:
-
-    # breakpoint()
-
-    # And run the script. You'll be dropped into a debugger where you can
-    # print() variables and see what's going on.
-    for schdoc_file in schdoc_files_in_repo:
-        if schdoc_file.path in schdoc_files_in_proj:
-            # If the file is not yet generated, we'll retry a few times.
-            retry_count = 0
-            while True:
-                retry_count += 1
-
-                try:
-                    schdoc_file_json = schdoc_repo.get_generated_json(
-                        schdoc_file, ref=schdoc_branch
-                    )
-                    bom_rows.extend(
-                        extract_attributes_from_schdoc(
-                            schdoc_file_json, attributes_to_extract
-                        )
-                    )
-                    break
-                except NotYetGeneratedException:
-                    if retry_count > 5:
-                        print(f"Failed to get {schdoc_file.path}, skipping...")
-                        break
-
-                    print(f"Failed to get {schdoc_file.path}, retrying...")
-                    time.sleep(1)
-                    continue
-
-    if args.output_file is not None:
-        f = open(args.output_file, "w")
-        writer = csv.writer(f)
-    else:
-        writer = csv.writer(sys.stdout)
-
-    # If you're extracting more or fewer attributes, you can change the header
-    # row here.
-    writer.writerow(attributes_to_extract)
-    writer.writerows(bom_rows)
+    bom_rows = generate_bom_for_altium(
+        allspice,
+        repository,
+        prjpcb_file,
+        pcb_file,
+        attributes_mapper,
+        args.source_branch,
+    )
+    bom_rows = [
+        [
+            bom_row.description,
+            ", ".join(bom_row.designators),
+            bom_row.quantity,
+            bom_row.manufacturer,
+            bom_row.part_number,
+        ]
+        for bom_row in bom_rows
+    ]
```

Comment:
What does the for loop at the end of the assignment do?
```
 bom_rows = [
        [
            bom_row.description,
            ", ".join(bom_row.designators),
            bom_row.quantity,
            bom_row.manufacturer,
            bom_row.part_number,
        ]
        for bom_row in bom_rows
    ]
```

Reply:
This is a Python list comprehension https://python.land/deep-dives/list-comprehension

It's a fancy way to create a list from some computation on another list.